resource/ui/ts/controllers/rest_explorer.ts

```diff
@@ -1,53 +1,306 @@
 // source: controllers/rest_explorer.ts
-/// <reference path="../typings/angularjs/angular.d.ts" />
-
-// Author: Andrew Bonventre (andybons@gmail.com)
-var crApp = angular.module('cockroach');
-crApp.controller('RestExplorerCtrl', ['$scope', '$http', function(scope, http) {
-  scope.kvCounterVal = 0;
-  scope.responseLog = [];
-  scope.clearResponseLog = function(e) {
-    scope.responseLog = [];
-  };
-  scope.requestPending = false;
-  scope.handleClick = function(e) {
-    e.preventDefault();
-    var method = e.target.getAttribute('data-method');
-    var endpoint = e.target.getAttribute('data-endpoint');
-    if (endpoint == '/kv/rest/range') {
-      endpoint += '?start=' + encodeURIComponent(scope.kvRangeStart);
-      if (!!scope.kvRangeEnd) {
-        endpoint += '&end=' + encodeURIComponent(scope.kvRangeEnd);
-      }
-    } else if (!!scope.kvKey) {
-      endpoint += scope.kvKey;
-    }
-    var data = '';
-    if (!!scope.kvValue) {
-      data = scope.kvValue;
-    }
-    var req = {
-      method: method,
-      url: endpoint,
-      headers: {
-        'Content-Type': 'text/plain; charset=UTF-8'
-      },
-      data: data
-    };
-    if (endpoint.indexOf('/kv/rest/counter/') != -1) {
-      req.headers = {
-        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+/// <reference path="../typings/mithriljs/mithril.d.ts" />
```

Comment:
In a nutshell: no.

Firstly, there is now no template file; angular was rendering this page with the rest_explorer.html template, which was removed in a subsequent PR.  Now all information about the page is in a single file.  The angular code here was only handling the json request, all of the structure was in the template file.

Secondly, the Typescript is definitely part of it.  This could have been written with far less code in pure javascript, but instead we've very clearly defined our Model data and are using cleanly scoped, typed controllers; this is extremely clean.  I also went a bit overboard in the case of this file in order to establish some patterns from writing mithril in Typescript.

Finally, the remaining verbosity means that there is very little magic in this framework.  Other SAP frameworks can be very succinct - Ember was very efficient in terms of lines of code.  It's not a problem until you need to debug something.